### ISSUE TITLE
Web3Container.js not mounting component for TestRPC/Ganache-CLI

### ISSUE THREAD

**Issue by lightswarm124** [adrianmcli/truffle-next#18]
Trying to deploy and test the example web app locally on my computer. I've set up a TestRPC instance on "localhost:8545", and changed the default "localProvider" to the same port number on the getWeb3.js file. The error is caught in at Web3Container.js with the async function "componentDidMount", and no error log showed up on console when running "npm run dev".  Not sure what configuration is missing or incorrect

**Comment by adrianmcli** [comment]
@lightswarm124 

Ah shoot, I should've made the port configurable. But it's changed to 9545 instead of 8545 in light of `truffle develop`'s default port. When v4 of Truffle came out, I made the decision to no longer support TestRPC, but clearly this was a mistake.

I'll put out a fix for this soon. In the mean time, you can change [this line](https://github.com/adrianmcli/truffle-next/blob/master/client/lib/getWeb3.js#L6) from:

```js
const localProvider = `http://localhost:9545`
```

```js
const localProvider = `http://localhost:8545`
```

Let me know if this temp fix works for you.

**Comment by adrianmcli** [comment]
Todo for myself:

- [ ] Allow user to configure `localProvider` URL and port.
- [ ] Make sure an informative error message pops up if we are not able to find the `localProvider`.

**Comment by lightswarm124** [comment]
It still does not connect through Web3. I think it may be due to Web3 1.0.0 adding callbacks. I noticed in the getWeb3.js file, it seems to use Web3 0.x API calls

**Comment by adrianmcli** [comment]
Hmm, that's strange. I've been testing it with web3 1.0.0-beta for many weeks now.

**Comment by adrianmcli** [comment]
Can you try to run testrpc like this:

```
testprc --gasLimit 6721975 --gasPrice 100000000000
```

[Source](https://github.com/trufflesuite/truffle/issues/660#issuecomment-343066784)

**Comment by lightswarm124** [comment]
Contracts are all deploying properly to testrpc with the new settings. Both the /dapp and /accounts pages give the alert saying "Failed to load web3, accounts, or contract. Check console for details". Could it either be the new WebsocketProvider replacing HttpProvider, or perhaps there are some IPC settings?

**Comment by adrianmcli** [comment]
I haven't tested it with testrpc in a while. I'll have to do it later today. Sorry for the delay.

**Comment by adrianmcli** [comment]
@lightswarm124 hmm I can't reproduce. It's working perfectly fine for me. Are you sure you've changed the line 6 of `getWeb3.js`?:

```
const localProvider = `http://localhost:8545`
```

**Comment by lightswarm124** [comment]
I've cloned a fresh instance of the repository on a new Ubuntu16 VM (separate from my main machine) with the port number change. Same error

**Comment by adrianmcli** [comment]
@lightswarm124 which error is that? Can you paste the error that's coming from the console?

**Comment by adrianmcli** [comment]
Also, instead of cloning the repository, have you tried `truffle unbox adrianmcli/truffle-next`? I just did a fresh `unbox` right now and everything is working fine with testrpc.

**Comment by lightswarm124** [comment]
I'm using "npm run dev", aka "next dev". The console only shows what pages are being built and how fast did the pages get built

**Comment by lightswarm124** [comment]
truffle unbox adrianmcli/truffle-next is what i used to get the repo

**Comment by adrianmcli** [comment]
1. Run `truffle unbox adrianmcli/truffle-next`, wait for it to finish.
2. `truffle migrate`
3. `cd client` and `vi lib/getWeb3.js` and change line 6 to use port 8545 instead of 9545
4. `npm run dev` or `yarn dev`
5. Run `testprc --gasLimit 6721975 --gasPrice 100000000000`
6. Navigate to `localhost:3000/accounts` and you should see a list of accounts. Make sure your MetaMask is off.

**Comment by adrianmcli** [comment]
@lightswarm124 btw, I was referring to the error from your browser console.

**Comment by lightswarm124** [comment]
Download the React DevTools for a better development experience: https://fb.me/react-devtools
react-dom.development.js:15338
[HMR] connected
client.js:82
unreachable code after return statement
[Learn More]
accounts:69625:4
unreachable code after return statement
[Learn More]
accounts:81046:4
unreachable code after return statement
[Learn More]
accounts:69625:4
unreachable code after return statement
[Learn More]
accounts:81046:4
No web3 instance injected, using Local web3.

**Comment by lightswarm124** [comment]
Error: SimpleStorage has not been deployed to detected network (network/artifact mismatch)
Stack trace:
deployed/<@http://localhost:3000/_next/1513129072140/page/accounts:97735:17
accounts:52068:17

**Comment by adrianmcli** [comment]
```
Error: SimpleStorage has not been deployed to detected network (network/artifact mismatch)
```

This is what you should be concerned with. It sounds like the network it is connecting to does not have the contract deployed. Did you do a `truffle migrate`? If your testrpc has previous stuff on it, you might need to do a `truffle migrate --reset`.

**Comment by lightswarm124** [comment]
Ya, I just got this to work. Thanks

**Comment by adrianmcli** [comment]
@lightswarm124 what was the issue? np

**Comment by lightswarm124** [comment]
I was resetting my testrpc a bit too much. Also, I'm not much of a web developer as much as a smart contract developer

**Comment by lightswarm124** [comment]
I think the main thing is the Metamask injection messing with the Web3 instance being fed, although it seems like Metamask is depreciating Web3 in the near future

**Comment by adrianmcli** [comment]
That's okay, although I would highly recommend using `truffle develop` instead of testrpc. Thanks for posting this issue, other people may have had the same problem.

MetaMask is supposed to be deprecating Web3 in the future, but we don't know when yet. I'm going to keep this project in line with the official Truffle Boxes for the mean time.
